lib/AST/NameLookup.cpp

```diff
@@ -1714,12 +1741,11 @@ void NominalTypeDecl::recordObjCMethod(AbstractFunctionDecl *method,
     return;
 
   if (auto *sf = method->getParentSourceFile()) {
-    if (vec.size() == 1) {
-      // We have a conflict.
-      sf->ObjCMethodConflicts.push_back(std::make_tuple(this, selector,
-                                                        isInstanceMethod));
-    } if (vec.empty()) {
+    if (vec.empty()) {
       sf->ObjCMethodList.push_back(method);
+    } else if (!isa<ProtocolDecl>(this) || !isAnAsyncAlternative(method, vec)) {
+      // We have a conflict.
+      sf->ObjCMethodConflicts.insert({ this, selector, isInstanceMethod });
```

Comment:
For those following along at home, I believe that the behavior change addressed in #59743 was caused by us now inserting conflicts into `ObjCMethodConflicts` when `vec.size() >= 1` instead of only when `vec.size() == 1`. When a method had both completion handler and async variants, `vec.size()` would start off at `2` and there would be no way for a user-defined conflicting method to ever add an entry to `ObjCMethodConflicts`.